converter/internal/otelcolconvert/converter_tailsamplingprocessor.go

```diff
@@ -0,0 +1,237 @@
+	block := common.NewBlockWithOverride([]string{"otelcol", "processor", "tail_sampling"}, label, args)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block)),
+	)
+
+	state.Body().AppendBlock(block)
+	return diags
+}
+
+func toTailSamplingProcessor(state *state, id component.InstanceID, cfg *tailsamplingprocessor.Config) *tail_sampling.Arguments {
+	var (
+		nextTraces = state.Next(id, component.DataTypeTraces)
+	)
+
+	testEncode := encodeMapstruct(cfg.PolicyCfgs[0])
+	spew.Dump(testEncode)
+
+	return &tail_sampling.Arguments{
+		PolicyCfgs:              toPolicyCfgs(cfg.PolicyCfgs),
+		DecisionWait:            cfg.DecisionWait,
+		NumTraces:               cfg.NumTraces,
+		ExpectedNewTracesPerSec: cfg.ExpectedNewTracesPerSec,
+		Output: &otelcol.ConsumerArguments{
+			Traces: toTokenizedConsumers(nextTraces),
+		},
+	}
+}
+
+func toPolicyCfgs(cfgs []tailsamplingprocessor.PolicyCfg) []tail_sampling.PolicyConfig {
+	var out []tail_sampling.PolicyConfig
+	for _, cfg := range cfgs {
+		out = append(out, tail_sampling.PolicyConfig{
+			SharedPolicyConfig: toSharedPolicyConfig(cfg),
+			CompositeConfig:    toCompositeConfig(cfg.CompositeCfg),
+			AndConfig:          toAndConfig(cfg.AndCfg),
+		})
+	}
+	return out
+}
+
+func toSharedPolicyConfig(cfg tailsamplingprocessor.PolicyCfg) tail_sampling.SharedPolicyConfig {
+	return tail_sampling.SharedPolicyConfig{
+		Name:                   cfg.Name,
+		Type:                   string(cfg.Type),
+		LatencyConfig:          toLatencyConfig(cfg.LatencyCfg),
+		NumericAttributeConfig: toNumericAttributeConfig(cfg.NumericAttributeCfg),
+		ProbabilisticConfig:    toProbabilisticConfig(cfg.ProbabilisticCfg),
+		StatusCodeConfig:       toStatusCodeConfig(cfg.StatusCodeCfg),
+		StringAttributeConfig:  toStringAttributeConfig(cfg.StringAttributeCfg),
+		RateLimitingConfig:     toRateLimitingConfig(cfg.RateLimitingCfg),
+		SpanCountConfig:        toSpanCountConfig(cfg.SpanCountCfg),
+		BooleanAttributeConfig: toBooleanAttributeConfig(cfg.BooleanAttributeCfg),
+		OttlConditionConfig:    toOttlConditionConfig(cfg.OTTLConditionCfg),
+		TraceStateConfig:       toTraceStateConfig(cfg.TraceStateCfg),
+	}
+}
+
+func toCompositeConfig(cfg tailsamplingprocessor.CompositeCfg) tail_sampling.CompositeConfig {
+	return tail_sampling.CompositeConfig{
+		MaxTotalSpansPerSecond: cfg.MaxTotalSpansPerSecond,
+		PolicyOrder:            cfg.PolicyOrder,
+		SubPolicyCfg:           toSubPolicyConfig(cfg.SubPolicyCfg),
+		RateAllocation:         toRateAllocationConfig(cfg.RateAllocation),
+	}
+}
+
+func toSubPolicyConfig(cfgs []tailsamplingprocessor.CompositeSubPolicyCfg) []tail_sampling.CompositeSubPolicyConfig {
+	var out []tail_sampling.CompositeSubPolicyConfig
+	for _, cfg := range cfgs {
+		out = append(out, tail_sampling.CompositeSubPolicyConfig{
+			SharedPolicyConfig: tail_sampling.SharedPolicyConfig{
```

Comment:
Should `cfg.AndCfg` be converted here? (CompositeSubPolicyCfg has a field for AndCfg)

Reply:
yes! really good catch on that one.